### PR TITLE
Add API Usage docs

### PR DESCRIPTION
**Pull request by prudrabhat**
## Description
 
Add Usage docs for MobileCore and Identity similar to [iOS repo](https://github.com/adobe/aepsdk-core-ios/tree/main/Documentation/Usage)

## Related Issue
#280

### CODE REVIEW

File: docs/Usage/Identity.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+##### Kotlin
+
+```kotlin
+val identityExtensionVersion: String = Identity.extensionVersion()
+```
+
+
+#### Registering Identity extension
+`Identity.EXTENSION` represents a reference to the IdentityExtension class that can be used to register
```
Comment (praveek) [inline docs/Usage/Identity.md:23]: Can you reword this sentence?

File: docs/Usage/Identity.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+```
+
+##### Kotlin
+
+```kotlin
+val identityExtensionVersion: String = Identity.extensionVersion()
+```
+
+
+#### Registering Identity extension
```
Comment (praveek) [inline docs/Usage/Identity.md:22]: Can you also move this to the top?

File: docs/Usage/Identity.md (hunk 1 above)
Comment (praveek) [reply docs/Usage/Identity.md:23]: `Identity.EXTENSION` represents a reference to the `IdentityExtension` class that can be registered with `MobileCore` via its `registerExtensions` api.

File: docs/Usage/Identity.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
+# Identity API Usage
+
+This document details the APIs provided by Identity extension, along with sample code snippets on how to properly use the APIs.
```
Comment (praveek) [inline docs/Usage/Identity.md:3]: The document lists

File: docs/Usage/MobileCore.md [hunk 4]
```diff
@@ -0,0 +1,549 @@
+# Mobile Core API Usage
+
+This document details the APIs provided by MobileCore, along with sample code snippets on how to properly use the APIs.
```
Comment (praveek) [inline docs/Usage/MobileCore.md:3]: This document lists

File: docs/Usage/MobileCore.md [hunk 5]
```diff
@@ -0,0 +1,549 @@
...
+```
+
+##### Kotlin
+
+```kotlin
+val coreVersion: String = MobileCore.extensionVersion()
+```
+
+
+#### Registering your Application with the MobileCore
```
Comment (praveek) [inline docs/Usage/MobileCore.md:37]: Can you reword this to something like `Initializing MobileCore`?

File: docs/Usage/README.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# API Usage
+
+The following documents provide sample code snippets for using the public APIs provided by the components in aepsdk-core-android.
+
+- [MobileCore](./MobileCore.md)
+- [Identity](./Identity.md)
```
Comment (praveek) [inline docs/Usage/README.md:6]: Can you also add 2 more pages for Signal and Lifecycle which talks about Registration and extensionVersion APIs

**Review by praveek — CHANGES_REQUESTED** [review]

File: docs/Usage/MobileCore.md (hunk 5 above)
Comment (prudrabhat) [reply docs/Usage/MobileCore.md]: Done. Reworded the title and moved to before the `Registering extensions and starting the SDK` section. I also moved `getApplication()` along with this.

File: docs/Usage/README.md (hunk 6 above)
Comment (praveek) [reply docs/Usage/README.md:6]: Can you also link Lifecycle & Signal pages from this page?

Comment (prudrabhat) [reply docs/Usage/README.md:6]: Whops! Added them now. 😅

**Review by praveek — APPROVED** [review]
